### ISSUE TITLE
[java] InsufficientStringBufferDeclaration doesn't handle non-constant conditionals

### ISSUE THREAD

**Issue by jsotuyod** [pmd/pmd#4888]
**Affects PMD Version:** 
7.0.0

**Rule:**

[InsufficientStringBufferDeclaration](https://pmd.github.io/latest/pmd_rules_java_performance.html#insufficientstringbufferdeclaration)

**Description:**

When reducing the expression to compute the size, the rule fails to process non-constant conditions. But in doing so, misses that some branches may effectively allocate too small a buffer.

Ideally, these expressions would be reduced to retrieve the **minimum possible value** to analyze, as the rule checks against the minimum number of appended chars expected.

**Code Sample demonstrating the issue:**

```java
public class ConditionalExpression {
    public void bar(int arg) {
        StringBuffer sb = new StringBuffer(arg > 1 ? 500 : 2);  // Should report a warning at this line, 2 is too small
        sb.append('a');
        sb.append('a');
        sb.append('a');
    }
}
```

**Expected outcome:**

PMD should report a violation at line 3, but doesn't. This is a false-negative.

**Running PMD through:** *[CLI | Ant | Maven | Gradle | Designer | Other]*
Any
